### ISSUE TITLE
Inference parameters

### ISSUE THREAD

**Issue by emarsh25** [gcorso/DiffDock#72]
Could I get more specific explanations for what the inference.py parameters are? I see the descriptions in the file but I am still not clear on what all of them are doing.

**Comment by RJ3** [comment]
Some of the integer parameters:
`batch_size` : this appears to be a standard DNN batch size. Smaller the batch, the less accuracy the estimate of the gradient will be. However, if the batch is set too large, it will overburden the VRAM of the GPU. The authors have an A100 which has 80 GB which is why I think they were defaulting to an unspecified value of 32. Realistically for most people this should be much less. I have an A6000 with 48 GB and for my models I set the batch size to 8.  The jupyter notebook example uses a value of 6.  

`samples_per_complex` : This parameter sets the number of total samples that are generated and corersponds to the number of ranked models in the results directory. The unspecified default is 10 but the sample instructions use 40. I usually look at the ranked confidence scores of the output and increase this number if I think it hasn't plateaued or converged. I am not sure if this is correct way of interpreting it.  

`inference_steps` : number of denoising steps, this refers to the denoising process of a Diffusion Generative Model  

`actual_steps` : number of denoising steps that are actually performed. Unclear how this is different than inference_steps above. The unspecified default is None, but the example instructions uses a value of 18 for it.

**Comment by gcorso** [comment]
Thanks @RJ3 for the answer. We run on A6000 48GB GPUs with `batch_size` of 10, but it may depend on the size of your largest proteins. `inference_steps` divides the total diffusion process into multiple steps, when `actual_steps` is specified to be a value less than `inference_steps` the last few steps of the diffusion (small noise scales where the model tends to overfit to the training data) are skipped. `no_final_noise` does not add the diffusion noise after the final step.
